texastribune/donations
# Remove CSS from webpack bundle

#### What's this PR do?

Adds cache busting of CSS files outside of Webpack

#### Why are we doing this? How does it help us?

Currently, we are running our CSS through a custom task (for scss compiling and optimizing) and Webpack (for cache busting). To cut down on running our CSS through 2 processes, we're making all the logic happen (compiling, optimizing, and cache busting) in the custom task.

TL;DR
Old way: Custom task + Webpack = final CSS
New way: Custom task = final CSS

#### How should this be manually tested?

_Test the build_
`yarn` then `make restart`
You should not see any red Webpack warnings nor should you see any CSS files referenced in the bundle summary.

![Screen Shot 2019-05-13 at 12 15 57 PM](https://user-images.githubusercontent.com/2974713/57640785-f1841000-7578-11e9-90f0-ff033b988762.png)

_Test the output_
Everything look normal here?
[http://local.texastribune.org/donate](http://local.texastribune.org/donate)
[http://local.texastribune.org/business](http://local.texastribune.org/business)
[http://local.texastribune.org/circle](http://local.texastribune.org/circle)
[http://local.texastribune.org/blastform](http://local.texastribune.org/blastform)
[http://local.texastribune.org/blast-promo](http://local.texastribune.org/blast-promo)
[http://local.texastribune.org/404example](http://local.texastribune.org/blast-404example)


#### How should this change be communicated to end users?

#### Are there any smells or added technical debt to note?
Just a tricky note that this should be accompanied with a [ds-toolbox release](https://github.com/texastribune/ds-toolbox/pull/17).

#### What are the relevant tickets?
Not directly, but should be helpful for future CSS iterations for UMP and otherwise.

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Updated the documentation/wiki? *(our CSS instructions in general probably need some updating. Adding as a todo)*

#### TODOs / next steps:

* [ ] *Add updated process to our Front-end wiki*
